src/nsrt_learning/segmentation.py

```diff
@@ -46,6 +51,47 @@ def _switch_fn(t: int) -> bool:
     return _segment_with_switch_function(trajectory, _switch_fn)
 
 
+def _segment_with_oracle(trajectory: GroundAtomTrajectory) -> List[Segment]:
+    """Segment a trajectory using oracle NSRTs.
+
+    If options are known, just uses _segment_with_option_changes().
+
+    Otherwise, starting at the beginning of the trajectory, keeps track of
+    which oracle ground NSRTs are applicable. When any of them have their
+    effects achieved, that marks the switch point between segments.
+    """
+    traj, all_atoms = trajectory
+    if not traj.actions or traj.actions[0].has_option():
```

Comment:
the first condition here is weird -- is it covering an edge case?

Reply:
nvm not needed